cpc.py

```diff
@@ -114,33 +212,144 @@ def configure_optimizers(self):
-        dataset = UnlabeledImagenet(self.hparams.data_dir,
-                                    nb_classes=self.hparams.nb_classes,
-                                    split='train',
-                                    transform=train_transform)
-
-        loader = DataLoader(
-            dataset=dataset,
-            batch_size=self.hparams.batch_size,
-            pin_memory=True,
-            drop_last=True,
-            num_workers=16,
-        )
+        if self.hparams.dataset == 'cifar10':
+            train_transform = cpc_transforms.CPCTransformsCIFAR10().train_transform
+
+        elif self.hparams.dataset == 'stl10':
+            stl10_transform = cpc_transforms.CPCTransformsSTL10Patches(
+                patch_size=self.hparams.patch_size,
+                overlap=self.hparams.patch_overlap
+            )
+            train_transform = stl10_transform.train_transform
+
+        elif self.hparams.dataset == 'imagenet128':
+            train_transform = cpc_transforms.CPCTransformsImageNet128Patches(self.hparams.patch_size, overlap=self.hparams.patch_overlap)
+            dataset = UnlabeledImagenet(self.hparams.data_dir,
+                                        nb_classes=self.hparams.nb_classes,
+                                        split='train',
+                                        transform=train_transform)
+
+        loader = self.dataset.train_dataloader(self.hparams.batch_size, transforms=train_transform)
+        return loader
+
+    def val_dataloader(self):
+        if self.hparams.dataset == 'cifar10':
+            test_transform = cpc_transforms.CPCTransformsCIFAR10().test_transform
+
+        if self.hparams.dataset == 'stl10':
+            stl10_transform = cpc_transforms.CPCTransformsSTL10Patches(
+                patch_size=self.hparams.patch_size,
+                overlap=self.hparams.patch_overlap
+            )
+            test_transform = stl10_transform.test_transform
+
+        if self.hparams.dataset == 'imagenet128':
+            train_transform = cpc_transforms.CPCTransformsImageNet128Patches(self.hparams.patch_size, overlap=self.hparams.patch_overlap)
+            dataset = UnlabeledImagenet(self.hparams.data_dir,
+                                        nb_classes=self.hparams.nb_classes,
+                                        split='val',
+                                        transform=train_transform)
+
+        loader = self.dataset.val_dataloader(self.hparams.batch_size, transforms=test_transform)
 
         return loader
 
+    @staticmethod
+    def add_model_specific_args(parent_parser):
+        from test_tube import HyperOptArgumentParser
+        parser = HyperOptArgumentParser(parents=[parent_parser], add_help=False)
+        parser.add_argument('--online_ft', action='store_true')
+        parser.add_argument('--dataset', type=str, default='cifar10')
+
+        (args, _) = parser.parse_known_args()
+
+        cifar_10 = {
```

Comment:
I'm trying to get results on CIFAR-10. I think these should work no?
@olivierhenaff